.pem-evidence/pem_validator.py

```
@@ -0,0 +1,767 @@
+
+
+def check_push_bypass(terminals_dir: Path | None) -> list[Finding]:
+    """Scan terminal logs for forbidden push bypass patterns (--no-verify etc.)."""
+    findings: list[Finding] = []
+    if terminals_dir is None or not terminals_dir.is_dir():
+        return findings
+    for f in sorted(terminals_dir.glob("*.txt")):
+        try:
+            text = f.read_text(encoding="utf-8", errors="replace")
+        except Exception:
+            continue
```

Ruff (0.15.12)

[error] `try`-`except`-`continue` detected, consider logging the exception

(S112)